# Standardized-benchmarking-of-alternative-splicing-detection

**ASB.py** (alternative splicing benchmarking) is a standardized benchmarking tool designed to unify the output of multiple alternative splicing analysis software, including **MAJIQ**, **PSI-Sigma**, **SUPPA2**, and **rMATS**. By converting results into a unified format, ASB enables downstream comparison, evaluation, and integration of splicing events across different tools.

The unified formats for seven event types are as follow:

<img width="865" height="532" alt="image" src="https://github.com/user-attachments/assets/9d7ad677-f0f2-43e2-8b92-b7dde4d4cafb" />

**System Requirements**

Operating System: Linux, macOS, or Windows (64-bit)

Python Version: Python 3.9 or higher

Required Packages:
  - pandas >= 2.3.0
  - numpy >= 1.23.5
  - matplotlib >= 3.5.2
  - seaborn >= 0.13.2
  - matplotlib-venn >= 1.1.2
  - venn >= 0.1.3
  - upsetplot >= 0.9.0
  - scipy >= 1.13.1
  - pillow >= 9.2.0

Hardware Requirements:
  - Minimum 4 GB RAM
  - Recommended 8 GB RAM for large datasets


**Installation**

Install the required packages with:
```
bash
pip install -r requirements.txt
```
Install from Github
```
git clone https://github.com/wangduo-ux/Standardized-benchmarking-of-alternative-splicing-detection.git
cd Standardized-benchmarking-of-alternative-splicing-detection
```

**Input**

The input consists of the results generated by four software tools. You can use your preferred custom settings for running SUPPA2, rMATS, PSI-Sigma, and MAJIQ. These output results should be saved within folders named after each software, and inside each software folder, there should be subfolders named according to the sample names, as illustrated below:

```
Your input directory
│
├── SUPPA2/
│   └── lab1_M8/
│       └── *.dpsi, *.psivec
│
├── rMATS/
│   └── lab1_M8/
│       ├── A3SS.MATS.JCEC.txt
│       ├── fromGTF.MXE.txt
│       ├── fromGTF.novelSpliceSite.A3SS.txt
│       ├── fromGTF.SE.txt
│       └── ...
│
├── PSI-Sigma/
│   └── lab1_M8/
│       ├── Homo_sapiens.GRCh38.109.sorted.gtf.mapping.txt
│       ├── PSIsigma_r10_ir3.sorted.txt
│       ├── PSIsigma.groupb.txt
│       └── ...
│
└── MAJIQ/
    └── lab1_M8/
        ├── alt3and5prime.tsv
        ├── alt3prime.tsv
        ├── alt5prime.tsv
        └── ...
```

**Output**

After running ASB.py, a uniform_ID column will be added to the output files of each software in their respective folders. 
In addition, a folder named sample_name will be created under your input directory, containing comparison plots of the results from different software, including intersections of detected alternative splicing events, dPSI quantification consistency plots, and intersections of differential splicing events.

**Command and options for ASB.py**
```
python ASB.py -s SUPPA2 rMATS PSI-Sigma MAJIQ -e SE A5SS A3SS MX RI AF AL -g Homo_sapiens.GRCh38.109.gtf -i Your_input_directory -o Your_output_directory -sn Your_sample_name
```
List of options available:
```
-s | --software: space separated list of tools for benchmarking from the following list:SUPPA2, rMATS, PSI-Sigma, MAJIQ.

-o | --output-file: name of the output directory.

-i | --input: input directory containing the output results from different software. Inside this folder, each software’s output should be organized in a subfolder named after the software, i.e., SUPPA2, rMATS, PSI-Sigma, MAJIQ. Within each software folder, there should be subfolders named according to the sample names, i.e., lab1_M8.

-g | ----gtf: path of gene annotation file with GTF format.

-sn | --sample_name: the name of your sample. It should be the same as the name of subfolder in your input directory, i.e., lab1_M8.

-e | --event: space separated list of events to generate from the following list:

SE: Skipping exon (SE) events
A5SS: Alternative 5' (A5) splice sites
A3SS: Alternative 3' (A3) splice sites
MX: Mutually Exclusive (MX) exons
RI: Retained intron (RI)
AF: Alternative first exons
AL: Alternative last exons
```
**Example data**

We provided example data to evaluate alternative splicing (AS) event-level detection methods, derived from a single laboratory’s analysis of the Quartet reference materials.
"Example data for standardized benchmarking of four alternative splicing detection tools. figshare. Dataset. https://doi.org/10.6084/m9.figshare.30215368.v1". 
For analysis of the example datasets, the -sn parameter should be specified as M8-D6, F7-D6, or D5-D6.     
        
**Citation**

The scripts here are for the Manuscript titled "**Benchmarking RNA-seq with the Quartet Reference Materials to establish Best Practices for Accurate Alternative Splicing Detection**" by Wang et al.
